cli/pipeline/run.go

```diff
@@ -201,6 +200,7 @@ func runPipelineStep(pipeline *PipelineDefinition, step *PipelineDefinitionStep,
 				if podDefinition.needsVolume() {
 					deleteVolumeClaim(clientset, podDefinition, flags)
 				}
+				deleteAndWait(clientset, podDefinition, flags)
```

Comment:
I wonder if we should do the same when it fails? I.e. by default always delete BUT if you really want you pass a flag to not delete on fail?

Reply:
I'd argue - that if it fails you might want to maybe SSH into the box - try to get some more information out - why it failed - `OOM` type messages are not logged by paddle. 

I created this ticket a few months back: https://deliveroo.atlassian.net/browse/PRIC-449 - if we do this then I'm comfortable to delete all pods.

Reply:
I would prefer that this is not the default behaviour of paddle to be honest. So basically all staging and test environments automatically will no longer do this and production could in future be given a flag to leave the pod untouched. I am just wary that this problem is continue to pop but now slower ?

Reply:
Chatted offline with Pedro about this - I think there's still a few ❓ - but not blocking.